packages/core/src/bundle.ts

```diff
@@ -205,3 +207,45 @@ function buildAdjacencyList(bundle: Bundle): AdjacencyList {
 
   return adjacencyList;
 }
+
+/**
+ * Converts a resource with contained resources to a transaction bundle.
+ * This function is useful when creating a resource that contains other resources.
+ * Handles local references and topological sorting.
+ * @param resource The input resource which may or may not include contained resources.
+ * @returns A bundle with the input resource and all contained resources.
+ */
+export function convertContainedResourcesToBundle(resource: Resource & { contained?: Resource[] }): Bundle {
+  // Create a clone so we don't modify the original resource
+  resource = deepClone(resource);
+
+  // Create the simple naive bundle
+  const simpleBundle = {
+    resourceType: 'Bundle',
+    type: 'transaction',
+    entry: [{ resource }] as BundleEntry[],
+  } satisfies Bundle;
+
+  // Move all contained resources to the bundle
+  if (resource.contained) {
+    for (const contained of resource.contained) {
+      simpleBundle.entry.push({ resource: contained });
+    }
+    resource.contained = undefined;
+  }
+
+  // Make sure that all resources have an ID
+  // This is required for convertToTransactionBundle
+  for (const entry of simpleBundle.entry) {
+    if (entry.resource && !entry.resource.id) {
+      entry.resource.id = generateId();
+    }
+  }
+
+  // Convert to a transaction bundle
+  // This adds fullUrl and request properties to each entry
+  const transactionBundle = convertToTransactionBundle(simpleBundle);
+
+  // Reorder the bundle to ensure that contained resources are created before they are referenced
+  return reorderBundle(transactionBundle);
```

Comment:
I think `convertToTransactionBundle` calls `reorderBundle` already, so this call may be unnecessary